File: caesar.c
// Julius Caesar cipher!
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{
    // Checks if there are two arguments or strings
    if (argc == 2)
    {
        // Iterates for each character in argv 
        for (int i = 0; i < strlen(argv[1]); i++)
        {
            // Checks if every character in argv is digit
            if (isdigit(argv[1][i]))
            {
            }
            else
            {
                printf("Usage: ./caesar key\n");
                return 2;
            }
        }
        // Changes argv from character to integer
        int j = atoi(argv[1]);
        // Prompting for plaintext
        string pt = get_string("plaintext: ");
        printf("ciphetext: ");
        // Dummy protection to not exceed ASCII table
        while (j >= 26)
        {
            j = j - 26;
            
        }
        // Iterates each character in plaintext
        for (int k = 0; k < strlen(pt); k++)
        {
            // Ciphers uppercase letters
            if (pt[k] >= 'A' && pt[k] <= 'Z')
            {
                // Protection against going out of letters
                if (pt[k] + j > 'Z')
                {
                    printf("%c", pt[k] - 26 + j);
                }
                else
                {
                    printf("%c", pt[k] + j);
                }
            }
            // Ciphers lowercase letters
            else
            {
                if (pt[k] >= 'a' && pt[k] <= 'z')
                {
                    // Again, protection against going out of letters
                    if (pt[k] + j > 'z')
                    {
                        printf("%c", pt[k] - 26 + j);
                    }
                    else
                    {
                        printf("%c", pt[k] + j);
                    }
                }
                // Prints characters that aren't alphabetical
                else
                {
                    printf("%c", pt[k]);
                }
            }
        }
        printf("\n");
        return 0;
    }
    else
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }
}

File: caesar.py
# import cs50 get string
from cs50 import get_string
import sys

# check if there are two arguments (strings)
if len(sys.argv) != 2:
    print("Usage: 'python caesar.py key', where the key is a number")
    sys.exit([1])

# assign int for key and get plaintext
k = int(sys.argv[1])
p = get_string("plaintext: ")
print("ciphertext: ", end="")

# iterate (and print) each character in plaintext
for c in p:
    # only for alphabetic characters
    if c.isalpha():
        # shift c by key, substracting difference between ascii value and position in alphabet
        if c.isupper() == True:
            c = ((ord(c) - 65 + k) % 26) + 65
        else:
            c = ((ord(c) - 97 + k) % 26) + 97
        c = chr(c)
    print(c, end="")
print()
